src/index.ts

```diff
@@ -0,0 +1,489 @@
+import * as crypto from 'crypto'
+import * as ethUtil from 'ethereumjs-util'
+
+const bs58check = require('bs58check')
+const randomBytes = require('randombytes')
+const scryptsy = require('scrypt.js')
+const uuidv4 = require('uuid/v4')
+
+// parameters for the toV3() method
+
+interface V3Params {
+  kdf: string
+  cipher: string
+  salt: Buffer
+  iv: Buffer
+  uuid: Buffer
+  dklen: number
+  c: number
+  n: number
+  r: number
+  p: number
+}
+
+function mergeToV3ParamsWithDefaults(params?: Partial<V3Params>): V3Params {
```

Comment:
Just a note: This function's body can be replaced with:

```ts
return {
  ... v3Defaults,
  ...params
}
```

And given that it is so small and only used once, it could be inlined.